src/main/scala/NSQ.scala

```diff
@@ -1,127 +1,163 @@
+import com.twitter.algebird.{Monoid, Semigroup}
 import com.twitter.storehaus.ReadableStore
 import com.twitter.storehaus.algebra.{Mergeable, MergeableStore}
 import com.twitter.summingbird._
-import com.twitter.summingbird.batch.{Timestamp, BatchID}
+import com.twitter.summingbird.graph._
+import com.twitter.summingbird.batch.{Timestamp, BatchID, Batcher}
 import com.twitter.summingbird.online.{OnlineServiceFactory, MergeableStoreFactory}
 import com.twitter.summingbird.planner.DagOptimizer
-import com.twitter.util.{Closable, Future, Time}
+import com.twitter.util.{Closable, Future, Time, FuturePool, Promise}
 
+import scala.language.existentials
 
-class NSQ extends Platform[NSQ]{
+sealed trait NSQPlan {
+  /* Returns when finished */
+  def run(): Future[Unit]
+  /* Exceptions should be sent to the run Future */
+  def close(): Unit
+}
 
-  // These can be taken from Memory
-  type Sink[-T] = (T => Unit) // TODO maybe want to make this something w/ Future[Unit] so we can chain stuff
-  type Plan[T] = Stream[NSQWrappedValue[T]]
+object NSQPlan {
+  implicit def monoidNSQPlan: Monoid[NSQPlan] = new Monoid[NSQPlan] {
+    val zero = new NSQPlan { def run() = Future.Unit; def close() = () }
+    def plus(a: NSQPlan, b: NSQPlan) = new NSQPlan {
+      def run() = a.run().join(b.run()).unit
+      def close() = {
+        a.close()
+        b.close()
+      }
+    }
+  }
+}
 
-  private type Prod[T] = Producer[NSQ, T]
-  private type JamfMap = Map[Prod[_], Stream[NSQWrappedValue[_]]]
+case class SourcePlan[T](fp: FuturePool, src: NSQPushSource[T], receiver: Receiver[T]) extends NSQPlan {
+  private val closed = new Promise[Unit]()
+  private def consumer = Future {
+    val cons = src.buildNSQConsumer(receiver)
+    cons.start()
+    cons
+  }
 
-  // These can be taken from Storm
-  type Store[-K, V] = MergeableStoreFactory[(K, BatchID), V]
-  type Service[-K, +V] = ReadableStore[K, V]
+  def run() = for {
+    c <- consumer
+    _ <- closed
+  } yield c.close()
 
-  // And these are ours
-  type Source[T] = NSQPullSource[T]
+  def close() = closed.setValue(())
+}
 
+trait NSQStore[-K, V] {
```

Comment:
Nit: in the interest of readers following what's going on, maybe we could take nsq out of the name of this trait? I'm not quite sure I know what else I have in mind, but since it's not actually talking to nsq at all, I figure we could try to avoid indicating that it does